includes/sjson/writer.h

```diff
@@ -88,41 +84,8 @@ namespace sjson
 	//
 	// To resolve this, we use SFINAE template logic to deduce a real type if we pass a function
 	// and if we pass bool, the type will not exist and the function is stripped.
-	// Different compilers behave differently here:
-	// VS2015 does not have any issues with the std::function overloads and there are no ambiguity with it but
-	// it does not support the SFINAE logic either with a return type or template argument. It complains the
-	// function is already declared.
-	// VS2017 works with the std::function overloads for x86 and x64 but not with ARM. It also does not work
-	// with a SFINAE return type but it works with a template argument.
-	// GCC and clang work with a SFINAE return type or template argument.
-	// As such, on VS2015, we use the std::function overloads and on VS2017, GCC, and clang we use SFINAE
-	// with a template argument.
-	namespace impl
-	{
-		template <class Sig, class = void>
-		struct invoke {};
-
-		template <class F, class... Args>
-		struct invoke<F(Args...), decltype(void(std::declval<F>()(std::declval<Args>()...)))>
-		{
-			using type = decltype(std::declval<F>()(std::declval<Args>()...));
-		};
-
-		template <typename Sig, typename T, typename = void>
-		struct invoke_test : std::false_type {};
-
-		template <typename... Args, typename T>
-		struct invoke_test<void(Args...), T,
-						   decltype(void(typename invoke<T(Args...)>::type()))> : std::true_type
-		{};
-
-		template <typename Sig, typename T>
-		constexpr bool invokable()
-		{
-			return invoke_test<Sig, T>::value;
-		}
-	}
-#endif
+	template<typename F, typename ArgType>
+	struct HavingArgument : std::enable_if<std::is_constructible<std::function<void(ArgType)>, F>::value> {};
```

Comment:
This is a really clever fix. I had tried all manner of SFINAE to get this working without luck.
Amazing work, thanks!

Reply:
Thanks! Boost's is_invocable gave me the basic idea. I first tried half a dozen alternative ideas that didn't work.